### DISCUSSION TITLE
How to access ProjectHelper() in Component.vue

### DISCUSSION THREAD

**Discussion by jgulledge19** [TiagoSilvaPereira/vemto-issues#338] [Q&A]
I am making a custom plugin and trying to figure out how to access the ProjectHelper() so I can use the getModelsNamespace() and related methods. How can it be accessed?

**Comment by TiagoSilvaPereira** [comment]
Hi @jgulledge19 do you need these methods inside the template? If yes, you can use **this.projectHelper**

```
this.projectHelper.getModelsNamespace()
```

**Reply by jgulledge19** [reply]
@TiagoSilvaPereira thanks for the quick reply. No I am needing the projectHelper in more then just the templates. I have written a plugin and trying to understand how to leverage the existing code. Part of the plugin is to allow adding a sub folder/path to be defined uniquely per model, similar to this: https://github.com/TiagoSilvaPereira/vemto-issues/discussions/292 

Visual of the plugin so far:
<img width="922" alt="image" src="https://user-images.githubusercontent.com/671767/177369793-e2588f6f-399a-4e49-8be9-545837d8b0d1.png">

Then I have a custom Model.vemtl file that writes and maps the path and namespace as indicated. I have a simple JS object in the main.js that has a function to calculate the path/namespace for a passed model.

**Reply by jgulledge19** [reply]
This an overlap from https://github.com/TiagoSilvaPereira/vemto-issues/discussions/338. I would like to have a simple function inside of Main.js that can figure out a custom namespace for a model.
